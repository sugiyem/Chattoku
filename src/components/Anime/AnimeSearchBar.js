import { StyleSheet, View } from "react-native";
import React from "react";
import {
  ButtonText,
  SearchButton,
  SeparatedSearchInput
} from "../../styles/GeneralStyles";

const AnimeSearchBar = ({
  value,
  onChangeText,
  onPress = null,
  navigation
}) => {
  const onSearchClick = () => {
    if (onPress) {
      onPress();
      return;
    }
    navigation.push("AnimeResult", { search: value });
  };

  return (
    <View style={styles.searchBarContainer}>
      <SeparatedSearchInput
        value={value}
        onChangeText={onChangeText}
        placeholder="Search anime by original title"
        testID="searchInput"
      />
      <SearchButton onPress={onSearchClick} testID="searchButton">
        <ButtonText color="#ffffff">Search</ButtonText>
      </SearchButton>
    </View>
  );
};

export default AnimeSearchBar;

const styles = StyleSheet.create({
  searchBarContainer: {
    flexDirection: "row"
  }
});
