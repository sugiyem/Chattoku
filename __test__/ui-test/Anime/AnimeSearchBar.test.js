import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react-native";
import AnimeSearchBar from "../../../src/components/Anime/AnimeSearchBar";

describe("Test Anime search bar UI", () => {
  it("Renders correctly", () => {
    const tree = renderer.create(<AnimeSearchBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Can change input and press button", () => {
    const mockChangeText = jest.fn();
    const mockPressSearchButton = jest.fn();
    const mockValue = "";

    const { getByTestId } = render(
      <AnimeSearchBar
        search={mockValue}
        onChangeText={mockChangeText}
        onPressSearchButton={mockPressSearchButton}
      />
    );

    fireEvent.press(getByTestId("searchButton"));
    fireEvent.changeText(getByTestId("searchInput"));

    expect(mockChangeText).toHaveBeenCalled();
    expect(mockPressSearchButton).toHaveBeenCalled();
  });
});
