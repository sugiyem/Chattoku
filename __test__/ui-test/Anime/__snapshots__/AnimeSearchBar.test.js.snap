// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Anime search bar UI Renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <TextInput
    placeholder="Search anime by original title"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#000000",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "#000000",
          "flexBasis": 0,
          "flexGrow": 3,
          "flexShrink": 1,
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingBottom": 2,
          "paddingLeft": 2,
          "paddingRight": 2,
          "paddingTop": 2,
        },
      ]
    }
    testID="searchInput"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#483d8b",
        "borderRadius": 10,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      }
    }
    testID="searchButton"
  >
    <Text
      color="#ffffff"
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 14,
            "textAlign": "center",
          },
        ]
      }
    >
      Search
    </Text>
  </View>
</View>
`;
